framework/include/geomsearch/ElementPairInfo.h

```diff
@@ -0,0 +1,56 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef ELEMENTPAIRINFO_H
+#define ELEMENTPAIRINFO_H
+
+// MOOSE includes
+#include "Moose.h"
+#include "DataIO.h"
+
+// libmesh includes
+#include "libmesh/vector_value.h"
+#include "libmesh/point.h"
+
+// libMesh forward declarations
+namespace libMesh
+{
+  class Node;
+  class Elem;
+}
+
+/**
+ * This is the ElementPairInfo class.  This is a base class that
+ * stores information used for integration in element to element constraints.
+ */
+
+class ElementPairInfo
+{
+public:
+
+  ElementPairInfo(const Elem * elem,
+                  const std::vector<Point> & constraint_q_point,
+                  const std::vector<Real> & constraint_JxW,
+                  const Point & normal);
+
+  virtual ~ElementPairInfo();
+
+public:
```

Comment:
this already is public, do you mean protected here? (and no empty lines after access qualifiers, and no empty line between the doxygen comment and the class declaration).